src/transmission_manager.rs

```diff
@@ -0,0 +1,293 @@
+use ixa::{
+    context::Context,
+    define_person_property, define_person_property_with_default, define_rng,
+    error::IxaError,
+    global_properties::ContextGlobalPropertiesExt,
+    people::{ContextPeopleExt, PersonId, PersonPropertyChangeEvent},
+    random::ContextRandomExt,
+};
+use statrs::distribution::{ContinuousCDF, Exp, Poisson};
+
+use crate::contact::QueryContacts;
+use crate::parameters::Parameters;
+
+#[derive(Debug, Hash, Eq, PartialEq, Clone, Copy)]
+pub enum InfectiousStatus {
+    Susceptible,
+    Infectious,
+    Recovered,
+}
+
+define_rng!(TransmissionRng);
+
+define_person_property_with_default!(
+    InfectiousStatusType,
+    InfectiousStatus,
+    InfectiousStatus::Susceptible
+);
+
+fn evaluate_transmission(context: &mut Context, contact_id: PersonId, _transmittee_id: PersonId) {
+    // for now, we assume all transmission events are sucessful
+    // we have the contact_id as an argument to this fcn so that person-person pair transmission potential
+    // based on interventions can be evaluated to determine whether this transmission event is actually successful
+    if matches!(
+        context.get_person_property(contact_id, InfectiousStatusType),
+        InfectiousStatus::Susceptible
+    ) {
+        context.set_person_property(
+            contact_id,
+            InfectiousStatusType,
+            InfectiousStatus::Infectious,
+        );
+    }
+}
+
+fn infection_attempt(
+    context: &mut Context,
+    transmitee_id: PersonId,
+    gi: f64,
+    num_infection_attempts_remaining: f64,
+    next_infection_time_unif: f64,
+) -> Result<(), IxaError> {
+    // if the person still has people left to infect
+    if num_infection_attempts_remaining >= 1.0 {
```

Comment:
Why isn't this an integer?

Reply:
Annoyingly, randomly generating from `Pois` returns a float instead of an integer even though all sampled values are integral. Any thoughts on how to fix this without casting/is casting just the right solution here?

Reply:
I think casting is the solution here: `rng.sample(Poisson) as usize` for example